### DISCUSSION TITLE
[Run issue]: The dose cannot be calculated

### DISCUSSION THREAD

**Discussion by ntat252** [OpenTOPAS/OpenTOPAS#28] [Run-Problem] [answered]
### What was the issue?

I am performing a simulation to calculate doses for patients. I have DICOM CT and rtdose datasets. I created an input file to run topas, but the output file does not contain the dose.

I do it in 2 ways:
1) use file HUtoMaterialSchneider.txt
For this method, I run successfully but don't get the dose in the output file

2) use the material assignment file
s:Ge/Patient/ImagingToMaterialConverter = "MaterialTagNumber"
iv:Ge/Patient/MaterialTagNumbers = 3 1 2 3
sv:Ge/Patient/MaterialNames = 3 "TG186MeanFemaleSoftTissue" "TG186Water" "Air"

However, with the second method, I encountered the following error:
Topas is exiting due to a serious error.
The ImageCube contains the material tag number: -979
However this tag number does not appear in Ge/Patient/MaterialTagNumbers

I look forward to receiving help from everyone.
Thanks

### TOPAS category

Scoring

### Version

4.0

### What Operative System (OS) are you running on?

Linux

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Attestation

- [X] I testify that I have fully read the bug reporting guideline beforehand.

**Comment by tmasilela** [comment]
Hi, 

Could you please share your parameter files so that we can have a more detailed look?

All the best,

Thongchai

**Comment by ntat252** [comment]
Hi Thongchai,

I attached my parameter files below.

[test.txt](https://github.com/user-attachments/files/15846694/test.txt)
[Ir192_Flexisource.txt](https://github.com/user-attachments/files/15846696/Ir192_Flexisource.txt)
[TG186Materials_cervical.txt](https://github.com/user-attachments/files/15846698/TG186Materials_cervical.txt)
[HUtoMaterialSchneider.txt](https://github.com/user-attachments/files/15846699/HUtoMaterialSchneider.txt)

I look forward to receiving help from you.
Thu

**Reply by JoseRamosMendez** [reply]
From the parameter files, the position of the CT is set at 0,0,0; but the source y positions is quite far. Cloning the dose grid only sets the dimension and resolution, but not the location nor the orientation. You have to check the CT isocenter, and set the position accordingly.

**Comment by ntat252** [comment]
Can you recommend to me how to check the CT isocenter? I haven't experienced it yet.
Thanks

**Comment by ntat252** [comment]
I did a lot of research on google but still couldn't solve the problem.

**Comment by ntat252** [comment]
I followed your instructions but still don't see the dose. I observed it with VITORIA.
![image](https://github.com/OpenTOPAS/OpenTOPAS/assets/131126638/ed64d944-8584-4440-a9be-d4f0314e9612)

Thanks

**Reply by JoseRamosMendez** [reply]
Hi there,
You can see from the Victoria window that your patient is not centered around zero in the Y and Z directions. By default, OpenTOPAS places the DICOM image at Geant4's coordinates 0, 0, 0. Thus you need to use the TransX/Y/Z parameters to reposition your DICOM-CT. Please check the example [ViewAbdomen_rtdose](https://github.com/OpenTOPAS/OpenTOPAS/blob/master/examples/Patient/ViewAbdomen_rtdose.txt) for more insights.
Best!

**Comment by ntat252** [comment]
Hi,
Can you suggest me a way to determine the isocenter?
I tried writing python code myself but it doesn't seem right.
Thanks!

**Reply by JoseRamosMendez** [reply]
Are you comfortable with sharing the anonymized DICOM-CT and DICOM-RTPLAN files?

**Comment by ntat252** [comment]
I'm very happy for your help
This is a link containing anonymized patient data

https://drive.google.com/drive/folders/1dPkZJ70rtaQOa7aufrtyPsELHN27

Thanks

**Comment by ntat252** [comment]
Looks like the previous link has a problem. I send the data link below.

https://drive.google.com/drive/folders/1dPkZJ70rtaQOa7aufrtyPsELHN27Zhlb?usp=sharing
Best!

**Comment by JoseRamosMendez** [comment, marked as answer]
At the begining of the simulation you can see the DICOM positioning 

![Screenshot 2024-06-25 at 9 56 10 AM](https://github.com/OpenTOPAS/OpenTOPAS/assets/12536336/d600ae52-3884-4a65-b78c-4cb96be1f5f1)

But it is useful to use IEC coordinate system, and DicomOriginX/Y/Z parameters to set the DICOM positioning:

    dc:Ge/Patient/DicomOriginX = 0.0 mm
    dc:Ge/Patient/DicomOriginY = 0.0 mm
    dc:Ge/Patient/DicomOriginZ = 0.0 mm

These parameters update internally once the DICOM-CT is read, thus you can assign such values to the Patient component translation

    d:Ge/Patient/TransX = Ge/Patient/DicomOriginX cm
    d:Ge/Patient/TransY = Ge/Patient/DicomOriginY cm
    d:Ge/Patient/TransZ = Ge/Patient/DicomOriginZ cm

This file has an example of how to use them properly. 
[PatientSetup.txt](https://github.com/user-attachments/files/15974996/PatientSetup.txt)


> [!NOTE]
> You must to correct the source orientation according to your DICOM. It seems that the source geometry is perpendicular to its traveling on the Y axis
>

![Screenshot 2024-06-25 at 10 03 23 AM](https://github.com/OpenTOPAS/OpenTOPAS/assets/12536336/05f21e96-050a-4855-9bd9-e72517da2dd6)

**Reply by tracyfhy** [reply]
Dear Ramos,
I read the PatientSetup.txt you shared and had one question that I wanted to confirm with you.
I also posted a discussion about the problem that I am recently stuck in https://github.com/OpenTOPAS/OpenTOPAS/discussions/82 . I am hopefully looking forward to your insightful thoughts.
Thanks in advance.

Best,
Hongying

**Comment by ntat252** [comment]
Can you share with me the input file that you used to run in this case?

**Reply by JoseRamosMendez** [reply]
The input file is linked above "PatientSetup.txt"

**Reply by ntat252** [reply]
Thank you so much for your support!!

**Comment by ntat252** [comment]
Dear Dr. José Ramos-Méndez,
I once received your help on the TOPAS forum. I'm very happy and grateful
for that. However because I am a student who is getting used to research, I
still have many difficulties in understanding and implementing it. I am
writing this email hoping to receive further support from you.

I have some questions after studying the example file you showed me earlier
(PatientSetup.txt)

I have assigned the source as "CapsuleWorld" in the test.txt file. However,
in your PatientSetup.txt file is the source assignment "capsule". When I
choose 1 of 2, the results change. Can you help me explain this problem?

Best regards,
Thu
